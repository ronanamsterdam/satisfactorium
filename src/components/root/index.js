
import React from 'react';
import Layout from 'components/layout';

import Device   from 'components/global/device';
import Theme   from 'components/global/theme';

import Nav      from "components/nav"
import Footer   from 'components/footer';

import style from './style.module.less';

export default ({ element, props }) => {
  // props provide same data to Layout as Page element will get
  // including location, data, etc - you don't need to pass it
  return <div className={style.root}>
    <Nav/>
    <div className={style.rootContent}>
      <Layout {...props}>{element}</Layout>
      <Footer />
    </div>
    <Device />
    <Theme />
  </div>
}