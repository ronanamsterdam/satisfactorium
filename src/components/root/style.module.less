@import '~style/index';

.root {
  display: grid;
  grid-template-columns: 50px auto;
}

@media (max-width: @tablet-break) {
  .root {
      grid-template-columns: auto;
  }
}