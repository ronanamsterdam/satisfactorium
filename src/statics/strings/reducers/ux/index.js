export const DEVICE_FORM_FACTORS = {
    MOBILE:     'MOBILE',
    TABLET:     'TABLET',
    DESKTOP:    'DESKTOP',
}

export const THEMES = {
  LIGHT:     'LIGHT',
  DARK:     'DARK',
}

export const LOCALE_TYPE_ENG     = 'eng';
export const LOCALE_TYPE_JP      = '日本語';

export default [
    LOCALE_TYPE_ENG,
    LOCALE_TYPE_JP,
]