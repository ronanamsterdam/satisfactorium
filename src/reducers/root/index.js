
import actionTypes from 'statics/actions';

import ux,  {uxInitialState}  from './ux';

const initialState = {
  ux: uxInitialState,
};

export default function testReducer(state = initialState, action) {
    switch (action.type) {
        case actionTypes.LOCALE_SET_LOCALE:
        case actionTypes.LOCALE_SET_LOADED:
        case actionTypes.SET_THEME:
        case actionTypes.DEVICE_DIMENSIONS_SET:
        case actionTypes.DEVICE_USER_AGENT_SET:
        case actionTypes.DEVICE_FORM_FACTOR_SET:
            return {
                  ...state,
                  ux: ux(state.ux, action)
              }
        default:
            return state;
    }
}