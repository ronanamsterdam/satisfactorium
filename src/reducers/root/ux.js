import actionTypes   from 'statics/actions';

import {DEVICE_FORM_FACTORS, THEMES}    from 'statics/strings/reducers/ux';

export const deviceInitialState = {
    dimensions: null,
    agent:      null,
    factor:     DEVICE_FORM_FACTORS.DESKTOP,
}

export const uxInitialState = {
    theme: THEMES.LIGHT,
    device:     deviceInitialState,
}

const device = function(state = deviceInitialState, action) {
    switch(action.type) {
      case actionTypes.DEVICE_FORM_FACTOR_SET:
          return {
              ...state,
              factor: action.factor || DEVICE_FORM_FACTORS.DESKTOP
          }
      case actionTypes.DEVICE_USER_AGENT_SET:
          return {
              ...state,
              agent: action.agent
          }
      case actionTypes.DEVICE_DIMENSIONS_SET:
          return {
              ...state,
              dimensions: action.dimensions
          }
      default:
          return state;
    }
}

export default function ux(state = uxInitialState, action) {
    switch(action.type) {
        case actionTypes.DEVICE_DIMENSIONS_SET:
        case actionTypes.DEVICE_USER_AGENT_SET:
        case actionTypes.DEVICE_FORM_FACTOR_SET:
          return {
              ...state,
              device: device(state.device, action)
          }
        case actionTypes.SET_THEME: {
          return {
            ...state,
            theme: action.theme
          }
        }
        default:
            return state;
    }
}