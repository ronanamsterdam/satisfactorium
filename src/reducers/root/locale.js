import actionTypes         from 'statics/actions';

import * as supportedLocaleTypes from 'statics/strings/reducers/ux';

export const localeInitialState = {
    collection:[
        {
            name:       supportedLocaleTypes.LOCALE_TYPE_ENG,
            code:       'en-us',
            lang:       'en',
            tb:         false,
        },
        {
            name:       supportedLocaleTypes.LOCALE_TYPE_JP,
            code:       'ja',
            lang:       'ja',
            tb:         true,
        },
    ],
    updating: false,
    selected: {
        name:       supportedLocaleTypes.LOCALE_TYPE_ENG,
        code:       'en-us',
        lang:       'en',
        tb:         false,
    },
};


export default function locale(state = localeInitialState, action) {
    switch (action.type) {
        case actionTypes.LOCALE_SET_LOCALE:
            const selected = state.collection[action.idx];
            return {
                ...state,
                selected,
            }
        case actionTypes.LOCALE_SET_IS_UPDATING:
            return {
                ...state,
                updating: action.updating
            }

        default:
            return state;
    }
}