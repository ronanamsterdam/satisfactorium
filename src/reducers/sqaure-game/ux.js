import appActions   from 'statics/actions';

import {DEVICE_FORM_FACTORS}    from 'statics/strings/reducers/ux';

export const deviceInitialState = {
    dimensions: null,
    agent:      null,
    factor:     DEVICE_FORM_FACTORS.DESKTOP,
}

export const uxInitialState = {
    device:     deviceInitialState,
}

const device = function(state = deviceInitialState, action) {
    switch(action.type) {
        case appActions.DEVICE_FORM_FACTOR_SET:
            return {
                ...state,
                factor: action.factor || DEVICE_FORM_FACTORS.DESKTOP
            }
        case appActions.DEVICE_USER_AGENT:
            return {
                ...state,
                agent: action.agent
            }
        case appActions.DEVICE_DIMENSIONS_SET:
            return {
                ...state,
                dimensions: action.dimensions
            }
    default:
        return state;
    }
}

export default function ux(state = uxInitialState, action) {
    switch(action.type) {
        case appActions.DEVICE_DIMENSIONS_SET:
        case appActions.DEVICE_USER_AGENT_SET:
        case appActions.DEVICE_FORM_FACTOR_SET:
            return {
                ...state,
                device: device(state.device, action)
            }
        default:
            return state;
    }
}