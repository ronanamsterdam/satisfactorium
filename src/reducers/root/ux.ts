import * as actionNames from 'statics/actions';
import {UxTypes}  from 'actions/types';

import {THEMES}    from 'common/statics/theme';
import {DEVICE_FORM_FACTORS}    from 'common/statics/device';

import locale, {localeInitialState} from './locale';

export const deviceInitialState = {
    dimensions: null,
    agent:      null,
    factor:     DEVICE_FORM_FACTORS.DESKTOP,
}

export const uxInitialState = {
    theme:      THEMES.LIGHT,
    locale:     localeInitialState,
    device:     deviceInitialState,
}

const device = function(state = deviceInitialState, action: UxTypes) {
    switch(action.type) {
      case actionNames.DEVICE_USER_AGENT_SET:
          return {
              ...state,
              agent: action.agent
          }
      case actionNames.DEVICE_FORM_FACTOR_SET:
          return {
              ...state,
              factor: action.factor || DEVICE_FORM_FACTORS.DESKTOP
          }
      case actionNames.DEVICE_DIMENSIONS_SET:
          return {
              ...state,
              dimensions: action.dimensions
          }
      default:
          return state;
    }
}

export default function ux(state = uxInitialState, action: UxTypes) {
    switch(action.type) {
        case actionNames.DEVICE_DIMENSIONS_SET:
        case actionNames.DEVICE_USER_AGENT_SET:
        case actionNames.DEVICE_FORM_FACTOR_SET:
          return {
              ...state,
              device: device(state.device, action)
          }
        case actionNames.SET_THEME: {
          return {
            ...state,
            theme: action.theme
          }
        }
        case actionNames.LOCALE_SET_LOCALE: {
          return {
            ...state,
            locale: locale(state.locale, action)
          }
        }
        default:
            return state;
    }
}