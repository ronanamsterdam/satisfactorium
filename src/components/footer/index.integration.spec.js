import React from 'react';
import { Provider } from "react-redux"
import { act } from 'react-dom/test-utils';

import {uxInitialState} from 'src/reducers/root/ux'

import Footer from './index'

describe('Footer', () => {
    test('Footer locale is being fetched and rendered correctly', async () => {
      const store = mockStore({
        root:{ux: uxInitialState},
      });

      let  wrapper;

      // INFO: the way it must be done due to async set state update in locale fetch
      await act(async () => {
        wrapper = await mount(<Provider store={store}>
          <Footer />
        </Provider >);
      })
      // INFO: mandatory after async set state for new locale template vars to get updated
      wrapper.update()

      // locale fetch is done
      expect(wrapper.find('a')).toHaveLength(2);
      // locale fetch is done
      expect(wrapper.find('a').last().text()).not.toBeFalsy();
      expect(wrapper.html()).toMatchSnapshot();
    })

    test('locale switching works', async () => {

      const store = realStore();

      let  wrapper;

      // INFO: the way it must be done due to async set state update in locale fetch
      await act(async () => {
        wrapper = await mount(<Provider store={store}>
          <Footer />
        </Provider >);
      })
      // INFO: mandatory after async set state
      wrapper.update();

      // locale fetch is done
      expect(wrapper.find('a')).toHaveLength(2);
      // locale fetch is done
      const locale1LinkText = wrapper.find('a').last().text();
      expect(locale1LinkText).not.toBeFalsy();

      // updating locale to ja
      wrapper.find('button').last().simulate("click");
      await act(async () => await wrapper.update())
      wrapper.update();

      expect(wrapper.find('a')).toHaveLength(2);
      // new localized link text
      const locale2LinkText = wrapper.find('a').last().text();
      expect(locale2LinkText).not.toBeFalsy();
      expect(locale2LinkText).not.toEqual(locale1LinkText);
    })
  }
)