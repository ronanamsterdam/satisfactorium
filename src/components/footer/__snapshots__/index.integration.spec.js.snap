// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer locale is being fetched and rendered correctly 1`] = `"<footer class=\\"container\\"><div class=\\"content\\"><span>© 2020, Check out whole project on my  </span><span><a href=\\"https://github.com/ronanamsterdam/satisfactorium\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Github</a> or read <a href=\\"https://www.linkedin.com/in/romanzhyliov\\">About me</a></span></div><ul class=\\"settings\\"><li><ul class=\\"container\\"><li><button class=\\"selected\\">ENG</button></li><li><button class=\\"\\">日本語</button></li></ul></li><li></li></ul></footer>"`;
